tests/integration/managedclusteraddon_test.go

```diff
@@ -0,0 +1,146 @@
+	When("creating or updating ManagedClusterAddOn", func() {
+		It("should not return validation error", func() {
+			newMirrorPeer := mirrorPeer1.DeepCopy()
+			newMirrorPeer.Spec = multiclusterv1alpha1.MirrorPeerSpec{
+				Items: []multiclusterv1alpha1.PeerRef{
+					{
+						ClusterName: "cluster1",
+						StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+							Name:      "test-storagecluster",
+							Namespace: "test-namespace",
+						},
+					},
+					{
+						ClusterName: "cluster2",
+						StorageClusterRef: multiclusterv1alpha1.StorageClusterRef{
+							Name:      "test-storagecluster",
+							Namespace: "test-namespace",
+						},
+					},
+				},
+			}
+			var mcAddOn1 addonapiv1alpha1.ManagedClusterAddOn
+			var mcAddOn2 addonapiv1alpha1.ManagedClusterAddOn
+			managedcluster1 := clusterv1.ManagedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster1",
+				},
+			}
+			managedcluster2 := clusterv1.ManagedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster2",
+				},
+			}
+			ns1 := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster1",
+				},
+			}
+			ns2 := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster2",
+				},
+			}
+			By("creating MirrorPeer. Also, this should automatically create ManagedClusterAddOn", func() {
+				err := k8sClient.Create(context.TODO(), &managedcluster1, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &managedcluster2, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &ns1, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				err = k8sClient.Create(context.TODO(), &ns2, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				err = k8sClient.Create(context.TODO(), newMirrorPeer, &client.CreateOptions{})
+				Expect(err).NotTo(HaveOccurred())
+			})
+			By("polling for the created ManagedClusterAddOn", func() {
+				Eventually(func() error {
+					err := k8sClient.Get(context.TODO(), mcAddOn1LookupKey, &mcAddOn1)
+					if err != nil {
+						return err
+					}
+					return fmt.Errorf("Waiting on ManagedClusterAddOn %s/%s to get created", mcAddOn1LookupKey.Namespace, mcAddOn1LookupKey.Name)
+				}, 60*time.Second, 2*time.Second).ShouldNot(HaveOccurred())
+
+				Eventually(func() error {
+					err := k8sClient.Get(context.TODO(), mcAddOn2LookupKey, &mcAddOn2)
+					if err != nil {
+						return err
+					}
+					return fmt.Errorf("Waiting on ManagedClusterAddOn %s/%s to get created", mcAddOn2LookupKey.Namespace, mcAddOn2LookupKey.Name)
+				}, 60*time.Second, 2*time.Second).ShouldNot(HaveOccurred())
+			})
+			By("updating ManagedClusterAddOn", func() {
+				mcAddOn1.Spec.InstallNamespace = "new-test-namespace"
+				err := k8sClient.Update(context.TODO(), &mcAddOn1, &client.UpdateOptions{})
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
after you update, very that install namespace changed.

also, add a case for when  addon is deleted, it should be recreated.